reactor-core/src/main/java/reactor/core/publisher/SinkManyBestEffort.java

```diff
@@ -0,0 +1,385 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package reactor.core.publisher;
+
+import java.util.Objects;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicLongFieldUpdater;
+import java.util.concurrent.atomic.AtomicReferenceFieldUpdater;
+import java.util.stream.Stream;
+
+import reactor.core.CoreSubscriber;
+import reactor.core.Exceptions;
+import reactor.core.Scannable;
+import reactor.core.publisher.Sinks.Emission;
+import reactor.util.annotation.Nullable;
+import reactor.util.context.Context;
+
+/**
+ * @author Simon Baslé
+ */
+final class SinkManyBestEffort<T> extends Flux<T> implements Sinks.Many<T>, ContextHolder,
+                                                             Scannable {
+
+	private static final Inner[] EMPTY = new Inner[0];
+	private static final Inner[] TERMINATED = new Inner[0];
+
+	static final <T> SinkManyBestEffort<T> createBestEffort() {
+		return new SinkManyBestEffort<>(false);
+	}
+
+	static final <T> SinkManyBestEffort<T> createAllOrNothing() {
+		return new SinkManyBestEffort<>(true);
+	}
+
+	final boolean allOrNothing;
+
+	/**
+	 * Stores the error that terminated this sink, for immediate replay to late subscribers
+	 */
+	Throwable error;
+
+	volatile     Inner<T>[]                                               subscribers;
+	@SuppressWarnings("rawtypes")
+	static final AtomicReferenceFieldUpdater<SinkManyBestEffort, Inner[]> SUBSCRIBERS =
+			AtomicReferenceFieldUpdater.newUpdater(SinkManyBestEffort.class, Inner[].class, "subscribers");
+
+	SinkManyBestEffort(boolean allOrNothing) {
+		this.allOrNothing = allOrNothing;
+		SUBSCRIBERS.lazySet(this, EMPTY);
+	}
+
+	public Context currentContext() {
+		return Operators.multiSubscribersContext(subscribers);
+	}
+
+	@Override
+	public Stream<? extends Scannable> inners() {
+		return Stream.of(subscribers);
+	}
+
+	@Nullable
+	@Override
+	public Object scanUnsafe(Attr key) {
+		if (key == Attr.TERMINATED) return subscribers == TERMINATED;
+		if (key == Attr.ERROR) return error;
+
+		return null;
+	}
+
+	@Override
+	public Emission tryEmitNext(T t) {
+		Objects.requireNonNull(t, "tryEmitNext(null) is forbidden");
+
+		Inner<T>[] subs = subscribers;
+
+		if (subs == EMPTY) {
+			return Emission.FAIL_ZERO_SUBSCRIBER;
+		}
+		if (subs == TERMINATED) {
+			return Emission.FAIL_TERMINATED;
+		}
+
+		if (allOrNothing) {
+			long commonRequest = Long.MAX_VALUE;
+			for (Inner<T> sub : subs) {
+				long subRequest = sub.requested;
+				if (sub.isCancelled()) {
```

Comment:
"all cancelled" is not handled

Reply:
it does fall through to the second looping, but yeah I should check `length == cancelledCount` early in that loop too